controllers/opsworks.py

```diff
@@ -1,27 +1,108 @@
+import sys
 
 import boto.opsworks
-import logging
+from boto.exception import JSONResponseError
 
 from ec2 import EC2Controller
-import settings
+from exceptions import *
 
+import logging
 log = logging.getLogger()
 
 class OpsworksController(EC2Controller):
 
-    def __init__(self, *args, **kwargs):
-        self._opsworks = None
-        super(OpsworksController, self).__init__(*args, **kwargs)
-
     @property
     def opsworks(self):
-        if self._opsworks is None:
-            self._opsworks = self.create_opsworks_connection()
+        if not hasattr(self, '_opsworks'):
+            self._opsworks = self.create_opsworks_conn()
         return self._opsworks
 
-    def create_opsworks_connection(self):
+    def create_opsworks_conn(self):
         conn = boto.opsworks.connect_to_region(self.region,
             aws_access_key_id=self.aws_access_key_id,
             aws_secret_access_key=self.aws_secret_access_key
+            )
+        try:
+            conn.describe_my_user_profile()
+            log.debug("boto.opsworks connection to region %s established", self.region)
+            return conn
+        except JSONResponseError, e:
+            if e.status == '400':
+                log.error(e.message)
+                log.error("Unable to establish boto.opsworks connection. Check your aws credentials.")
+                sys.exit(1)
+            else:
+                raise
+
+    @property
+    def stack(self):
+        if not hasattr(self, '_stack'):
+            stacks = self.opsworks.describe_stacks()
+            try:
+                self._stack = next(x for x in stacks['Stacks'] if x['Name'] == self.prefix)
+            except StopIteration:
+               raise ClusterException(
+                   "Can't find a stack named {}".format(self.prefix)
+               )
+        return self._stack
+
+    @property
+    def opsworks_instances(self):
+        if not hasattr(self, '_opsworks_instances'):
+           self._opsworks_instances = self.opsworks.describe_instances(self.stack['StackId'])['Instances']
+        return self._opsworks_instances
+
+    @property
+    def instance_filter(self):
```

Comment:
I think we want to filter additionally for a null AutoScalingType: https://docs.aws.amazon.com/opsworks/latest/APIReference/API_Instance.html

We can't explictly start or stop `load` or `timer` instances, so this should ignore them entirely if they're on a stack.